### ISSUE TITLE
[Bug] Log-linear attention reads `level_scales` out of bounds, and drops a gate-gradient term, when the sequence length is not a multiple of 64

### ISSUE THREAD

**Issue by truong-v** [fla-org/flash-linear-attention#1142]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

`chunk_log_linear_attn` fixes the chunk size at `BT = 64`, so any sequence length that is not a multiple of 64 ends in a chunk whose trailing rows are dead. Three statements do not account for those rows.

**1. `chunkwise_fwd_kernel` reads `level_scales` past the end** (`chunk.py:171-172`). The gather is masked with the causal predicate only:

https://github.com/fla-org/flash-linear-attention/blob/033b19e81239b13971a410e55dd6c178d430b9d4/fla/ops/log_linear_attn/chunk.py#L171-L172

Every other load in the same loop carries the row-liveness mask `m_t`, computed two lines below. Without it, the dead rows still form addresses into `level_scales` and load from them — past the end of the tensor. The values are killed afterwards by the zero that `b_s` gets for dead rows, so this is silent whenever the memory behind `level_scales` happens to hold ordinary numbers.

**2. `chunkwise_bwd_kernel_diag` has the identical statement** (`chunk.py:989-990`), so the backward overruns the same tensor. Here it is *not* always silent: `b_dg -= tl.sum(b_dl * b_h, axis=0)` reduces down a column, over dead rows included, so a non-finite value behind `level_scales` reaches live gradients. With a NaN there, a `B=1, T=70, H=4, D=64` backward returns 384 non-finite `dk` entries, 1536 non-finite `dv` entries and 24 non-finite `dg` entries.

**3. `chunkwise_bwd_kernel_dkg` deposits the chunk-scalar gate gradient on a dead row** (`chunk.py:849-860`). It reads the gate at the chunk's last *live* token and then folds the result in at chunk-relative row `BT - 1`:

```python
last_idx = min((i_t + 1) * BT, T) - 1          # last LIVE token
b_g_last = tl.load(g + bos * H + last_idx * H + i_h)
...
b_dg_last *= tl.exp(b_g_last)
b_dg = tl.where(o_i < BT - 1, b_dg, b_dg + b_dg_last)   # chunk-RELATIVE row BT - 1
tl.store(p_dg, b_dg, mask=m_t)                          # which the mask then discards
```

Those are the same row only when the chunk is full; on a partial chunk the deposit row is dead and the masked store throws the term away. `chunkwise_fwd_kernel_h` and `chunkwise_bwd_kernel_dh` both rescale the state by the *last live* token's gate, so its gradient belongs on `dg[last_idx]`. As far as I can tell this one is currently latent rather than live — `chunkwise_bwd_kernel_dhg` writes `dg_last[i_t]` under `if i_t & (1 << ell)`, and for the last chunk of a sequence that add always runs on the first iteration of its descending loop, where the accumulator is still zero — so the pipeline hands this kernel a zero `dg_last` for exactly the chunk that can be partial. It is still an inconsistency inside the kernel, and it is one line from the same root cause.

Sites 1 and 2 arrived with the kernels (#524 and #577); site 3 with #577.

### Steps to Reproduce the Bug

```python
import numpy as np, torch
from fla.ops.log_linear_attn import chunk_log_linear_attn

B, T, H, D = 1, 70, 4, 64          # 70 = 64 + 6: the last chunk has 6 live rows
dev, dt_ = 'cuda', torch.float32
torch.manual_seed(42)
x  = torch.randn(B, T, H, D, dtype=dt_, device=dev)
dt = torch.nn.functional.softplus(torch.randn(B, T, H, dtype=torch.float32, device=dev) - 4)
a  = -torch.exp(torch.rand(H, dtype=torch.float32, device=dev))
q  = torch.randn(B, T, 1, D, dtype=dt_, device=dev)
k  = torch.randn(B, T, 1, D, dtype=dt_, device=dev)
v, g, do = (x * dt.unsqueeze(-1)), a * dt, torch.randn_like(x)

# level_scales at the front of a buffer whose tail is NaN, so a read past it is visible
L = int(np.ceil(np.log2(T))) + 1
buf = torch.full((B*T*H*L + 64*H*L,), float('nan'), dtype=dt_, device=dev)
ls  = buf[:B*T*H*L].view(B, T, H, L)
ls.copy_(torch.randn(B, T, H, L, dtype=dt_, device=dev))

for t in (q, k, v, g, ls):
    t.requires_grad_()
out, _ = chunk_log_linear_attn(q, k, v, g, ls)
(out * do).sum().backward()
for n, t in zip(('dq','dk','dv','dg','dl'), (q, k, v, g, ls)):
    print(n, int((~torch.isfinite(t.grad)).sum()), 'non-finite of', t.grad.numel())
```

```
dq 0 non-finite of 4480
dk 384 non-finite of 4480
dv 1536 non-finite of 17920
dg 24 non-finite of 280
dl 0 non-finite of 2240
```

The same shape with an ordinary `level_scales` is silent, but `compute-sanitizer --tool memcheck` with `PYTORCH_NO_CUDA_MEMORY_CACHING=1` reports 16 invalid 4-byte reads at `chunk.py:172` landing past `level_scales`' 8,960-byte allocation, after which the context dies; clamping only the forward moves the same 16 reads to `chunkwise_bwd_kernel_diag`.

### Expected Behavior

No access outside `level_scales`, and the chunk-scalar gate gradient deposited on the row whose gate it was scaled by.

### Environment Information

1. Torch: 2.13.0+cu130
2. Triton: 3.7.1
3. Commit tested: 033b19e8 (current main), Python 3.12, NVIDIA B200, driver CUDA 13.2
